gateway/api/services/file_storage.py

```diff
@@ -0,0 +1,125 @@
+"""
+This file stores the logic to manage the access to data stores
+"""
+import glob
+import logging
+import os
+from enum import Enum
+
+from django.conf import settings
+
+from utils import sanitize_file_path
+
+
+class WorkingDir(Enum):
+    """
+    This Enum has the values:
+        USER_STORAGE
+        PROVIDER_STORAGE
+
+    Both values are being used to identify in
+        FileStorage service the path to be used
+    """
+
+    USER_STORAGE = 1
+    PROVIDER_STORAGE = 2
+
+
+SUPPORTED_FILE_EXTENSIONS = [".tar", ".h5"]
+
+logger = logging.getLogger("gateway")
+
+
+class FileStorage:  # pylint: disable=too-few-public-methods
+    """
+    The main objective of this class is to manage the access of the users to their storage.
+
+    Attributes:
+        username (str): storgae user's username
+        working_dir (WorkingDir(Enum)): working directory
+        function_title (str): title of the function in case is needed to build the path
+        provider_name (str | None): name of the provider in caseis needed to build the path
+    """
+
+    def __init__(
+        self,
+        username: str,
+        working_dir: WorkingDir,
+        function_title: str,
+        provider_name: str | None,
+    ) -> None:
+        self.file_path = None
+        self.username = username
+
+        if working_dir is WorkingDir.USER_STORAGE:
+            self.file_path = self.__get_user_path(function_title, provider_name)
+        elif working_dir == WorkingDir.PROVIDER_STORAGE:
```

Comment:
```suggestion
        elif working_dir is WorkingDir.PROVIDER_STORAGE:
```
In order to unify both conditions

Reply:
Ups, my fault haha

Reply:
Solved in [8cce843](https://github.com/Qiskit/qiskit-serverless/pull/1529/commits/8cce843be9a58728903a2af6a349f5ea44e6a714)